go/cmd/vtcombo/cli/main.go

```diff
@@ -189,17 +189,20 @@ func run(cmd *cobra.Command, args []string) (err error) {
 		cmd.Flags().Set("log_dir", "$VTDATAROOT/tmp")
 	}
 
+	ctx, cancel := context.WithCancel(cmd.Context())
+	defer cancel()
```

Comment:
This creates the toplevel context here and then uses it for example for creating the memory topo and other operations too. 

Once this cancels, it ensures that things like the topo and watchers etc. are closed in the right order.